### DISCUSSION TITLE
Multiple Thermostat

### DISCUSSION THREAD

**Discussion by florianbeck85** [Blackymas/NSPanel_HA_Blueprint#293] [Ideas]
Hi Blackymas,

thank you so much for this features. Do you plan to integrate more than 1 Thermostat per panel ?

**Comment by Blackymas** [comment]
We have it on the roadmap, but we don't yet know exactly how to implement it because it consumes additional memory on the ESPHome side

**Reply by florianbeck85** [reply]
how about to use the thermostat function to activate the heating when the temp in the room is under the reat temperature of the nspanel and activace the hvac when it is over the panel temp

**Comment by Churator** [comment]
It means I can't control both my underfloor heating AND Air-condition? each one is different climate entity.

**Comment by Blackymas** [comment]
@all 

we have discussed the issue today and an idea we will try to solve it via the buttonpage.

There will be a button climate with the current temp and a label and with long press you come to the appropriate climate page to control.

Think this is the best solution ;)

See also here: https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/78#issuecomment-1420269189

**Reply by nitruxa** [reply]
I've played around with the climate control, as I had to do a separate one for my sauna... and the problem I faced was that sauna is not gonna work with default 12-28 degrees :)

how I achieved it for now is that I've created a separate page for sauna thermostat, and updated the logic how thermostat circle is being rendered.

This is what I thought to implement for myself (maybe it'd be useful) as a next version:
1. create a general climate page not tied to any climate entity
2. when clicking on a certain climate entity, set a variable to be aware which entity is being controlled/read
3. thermostat cycle state indicator should be set:
3.1. directly from the ESP.yaml when controlled via NSPanel
3.2. from automation passing `temperature`, `min_temp` and `max_temp` from climate entity and calculate percentages to show within `send_thermostat_cycle_state` service
4. call climate service directly from the ESP and sending calculated temperature. 

P.S. I've asked a friend UX designer to update home page a bit to suit my needs more, we can also redesign a bit the thermostat page as well if needed 😉

**Reply by deejaybeam** [reply]
currently working on it. (multiple climate)
But this depends on a little bit more (just have a look at current dev-branch where I already have prepared someting to do that).
The current graph-design (hard work from @Blackymas ) and limitations of how nextion works, makes it hard to change the range of the climate-slider dynamically because it is created from 28 pics over 3 fields.... (limitation of nextion)
BUT from the first implementation in v2 you have the possibility to use the up/down Buttons on the climate page outside the "normal" range of a room-climate.
Anyways: If you have a solution how to map "real" min/max to the existing climate without breaking the solution for room-climate with no additional page in TFT you are welcome to contribute - thats open source! 
I´m curious what your UX designer can build with nextion-editor.
Would be awesome if this could be possible - also helps for fahrenheit degrees!

**Reply by nitruxa** [reply]
@deejaybeam do u use slack or other chats? would love to discuss a bit in person that thermostat thing :)

**Reply by Fjury** [reply]
I have also been messing around and made a few adjustments to the climate tft page and in blueprint.yaml.

now i can link to all buttonpages also climate entities, which work just like the original. Optimistic mode is also taken over. 

With normal heaters it works. It is still missing the addition for e.g. hvac_mode=cool to be able to control air conditioners sensibly. But this should be only an extension of the mui.

Please tell me where to put my two files.... never uploaded anything to github ;-)

**Reply by deejaybeam** [reply]
@Fjury multi-climate is ready in dev-branch
This changes many of the whole funtionality - have a look on it
It also changes all hold-functions, decrease EspHome-Flash-Size and brings more flexibility

**Comment by deejaybeam** [comment]
@nitruxa you can find me on discord: deejaybeam#0105

**Reply by Fjury** [reply]
you got an invitation from Grille#2900

**Comment by edwardtfn** [comment]
I'm closing this discussion as we have released v4.0 (#1057) and you can now have another thermostat connected to a custom button in the home page.
Please add a feature request for the missing things.

And thanks a lot for your support.
